cmd/autoheal/alert_worker_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/autoheal/pkg/alertmanager"
+	"github.com/openshift/autoheal/pkg/apis/autoheal"
+	batch "k8s.io/api/batch/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func TestPickAlert(t *testing.T) {
+	file := filepath.Join("..", "..", "testdata", "empty-config.yml")
+	healer, err := NewHealerBuilder().
+		ConfigFile(file).
+		Build()
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	defer runtime.HandleCrash()
+
+	alert := &alertmanager.Alert{
+		Status: "firing",
+		Labels: map[string]string{
+			"mylabel": "myvalue",
+		},
+	}
+
+	healer.alertsQueue.Add(alert)
+
+	// one alert in queue.
+	if healer.pickAlert() != true {
+		t.Errorf("Expected pickAlert to return true (i.e. there is an alert in alertQueue) but got false")
+	}
+}
+
+func TestStartHealing(t *testing.T) {
+	file := filepath.Join("..", "..", "testdata", "empty-config.yml")
+	healer, err := NewHealerBuilder().
+		ConfigFile(file).
+		Build()
+
+	healer.actionRunners[ActionRunnerTypeAWX] = FakeAWXActionRunner{}
+	healer.actionRunners[ActionRunnerTypeBatch] = FakeBatchActionRunner{}
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	// Test 0
+
+	alert0 := &alertmanager.Alert{
+		Status: "firing",
+		Labels: map[string]string{
+			"mylabel": "myvalue",
+		},
+	}
+
+	change0 := &RuleChange{
+		Type: watch.Added,
+		Rule: &autoheal.HealingRule{
+			ObjectMeta: meta.ObjectMeta{
+				Name: "test-rule",
+			},
+			Labels: map[string]string{
+				"mylabel": "myvalue",
+			},
+			AWXJob: &autoheal.AWXJobAction{
+				Template: "Test AWX JOB",
+			},
+		},
+	}
+
+	// Add the rule change to rulesCache
+	healer.processRuleChange(change0)
+
+	output := healer.startHealing(alert0)
+
+	expected := fmt.Sprintf("Fake AWXActionRunner called with rule '%s' and alert '%s'",
```

Comment:
Maybe also test if it was called with the correct AWXJob? And add another "dummy" rule to make sure rule matching works as intended?

Reply:
`healer_test.go` contains a bunch of rule matching tests so I didn't want to touch on that logic.
Also, I feel like this essentially does check that it is the correct rule (If you look at the`RunAction` implementation in the stubs `Fake*ActionRunner` at `healer_test.go`).

Reply:
@yaacov What do you think?